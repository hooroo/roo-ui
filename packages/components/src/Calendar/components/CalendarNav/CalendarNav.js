import React from 'react';
import PropTypes from 'prop-types';
import { themeGet } from 'styled-system';

import { Flex, NakedButton, Icon } from '../../../';

const Wrapper = Flex.extend`
  width: 100%;
  position: absolute;
  justify-content: space-between;
  padding: 0 ${themeGet('space.4')};
`;

const Button = NakedButton.extend`
  border-radius: ${themeGet('radii.rounded')};
  background: ${themeGet('colors.white')};
  color: ${themeGet('colors.grey.1')};
  box-shadow: ${themeGet('shadows.default')};

  &:hover,
  &:focus {
    outline: none;
    color: ${themeGet('colors.brand.primary')};
  }

  &:disabled {
    cursor: not-allowed;
    box-shadow: none;
    color: ${themeGet('colors.grey.2')};
  }
`;

const CalendarNav = ({ prevProps, nextProps }) => (
  <Wrapper>
    <Button {...prevProps}>
      <Icon name="chevronLeft" />
    </Button>

    <Button {...nextProps}>
      <Icon name="chevronRight" />
    </Button>
  </Wrapper>
);

CalendarNav.propTypes = {
  prevProps: PropTypes.shape().isRequired,
  nextProps: PropTypes.shape().isRequired,
};

export default CalendarNav;
