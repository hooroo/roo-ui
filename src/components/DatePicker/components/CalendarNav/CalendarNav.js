import React from 'react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import { themeGet } from '@styled-system/theme-get';

import { Flex, NakedButton, Icon } from 'components';

const Wrapper = styled(Flex)`
  width: 100%;
  position: absolute;
  justify-content: space-between;
  padding: 0 ${themeGet('space.4')};
`;

const Button = styled(NakedButton)`
  border-radius: ${themeGet('radii.rounded')};
  background: ${themeGet('colors.white')};
  color: ${themeGet('colors.greys.steel')};
  box-shadow: ${themeGet('shadows.default')};

  &:hover,
  &:focus {
    outline: none;
    color: ${themeGet('colors.brand.primary')};
  }

  &:disabled {
    cursor: not-allowed;
    box-shadow: none;
    color: ${themeGet('colors.greys.alto')};
  }
`;

Button.displayName = 'Button';

const CalendarNav = ({ prevProps, nextProps }) => (
  <Wrapper>
    <Button {...prevProps}>
      <Icon name="chevronLeft" />
    </Button>

    <Button {...nextProps}>
      <Icon name="chevronRight" />
    </Button>
  </Wrapper>
);

CalendarNav.propTypes = {
  prevProps: PropTypes.shape().isRequired,
  nextProps: PropTypes.shape().isRequired,
};

export default CalendarNav;
