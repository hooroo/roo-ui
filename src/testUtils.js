import React from 'react';
import { mount, shallow } from 'enzyme';

export const mountWithTheme = (tree, theme) => {
  const themedTree = React.cloneElement(tree, { theme });
  return mount(themedTree);
};

export const shallowWithTheme = (tree, theme) => {
  const themedTree = React.cloneElement(tree, { theme });
  return shallow(themedTree);
};
