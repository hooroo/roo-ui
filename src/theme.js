import { rem } from 'polished';
import range from 'lodash/range';

import * as icons from './defaultIcons';

const namedColors = {
  white: '#FFFFFF',
  brightSun: '#FBCB3B',
  red: '#E00E00',
  maroon: '#870000',
  blue: '#8DE2E0',
  lightBlue: '#E7F7F7',
  orange: '#ED710B',
  lightOrange: '#FCEBCD',
  green: '#35A509',
  lightGreen: '#DEF2DE',
};

const brand = {
  primary: '#E40000',
  secondary: '#8DE2E0',
  tertiary: '#D20000',
  quaternary: '#F9F3E9',
};

const greys = {
  charcoal: '#323232',
  steel: '#666666',
  alto: '#DADADA',
  porcelain: '#F4F5F6',
  dusty: '#999999',
  mineShaft: '#222222',
};

const ui = {
  error: namedColors.orange,
  errorBackground: namedColors.lightOrange,
  info: namedColors.blue,
  infoBackground: namedColors.lightBlue,
  link: namedColors.red,
  linkHover: namedColors.maroon,
  success: namedColors.green,
  successBackground: namedColors.lightGreen,
};

const colors = {
  brand,
  greys,
  ui,
  ...namedColors,
};

const alertStyles = {
  default: { bg: 'greys.porcelain' },
  error: {
    bg: 'ui.errorBackground',
    icon: {
      name: 'warning',
      color: 'ui.error',
    },
  },
  info: {
    bg: 'ui.infoBackground',
    icon: {
      name: 'info',
      color: 'greys.charcoal',
    },
  },
  success: {
    bg: 'ui.successBackground',
    icon: {
      name: 'checkCircle',
      color: 'ui.success',
    },
  },
};

const borders = [0, '1px solid', '2px solid', '3px solid'];

const breakpoints = ['40rem', '52rem', '64rem'];

const buttons = {
  default: {
    color: 'white',
    backgroundColor: 'greys.charcoal',
  },
  primary: {
    color: 'white',
    backgroundColor: 'brand.primary',
  },
};

const fontFamily = 'Ciutadella, sans-serif';

const fontSizes = {
  xs: rem('13px'),
  sm: rem('15px'),
  base: rem('16px'),
  lg: rem('18px'),
  xl: rem('22px'),
  '2xl': rem('26px'),
  '3xl': rem('36px'),
  '4xl': rem('46px'),
};

const fontWeights = {
  normal: 500,
  bold: 600,
};

const letterSpacings = {
  normal: '0',
  wide: '0.125em',
};

const lineHeights = {
  tight: 1,
  normal: 1.35,
  loose: 1.5,
};

const maxWidths = {
  default: rem('1280px'),
};

const mediaQueries = breakpoints.map(
  n => `@media screen and (min-width: ${n})`,
);

const opacity = {
  disabled: 0.7,
};

const radii = {
  none: 0,
  default: '4px',
  rounded: '1000px',
};

const shadows = {
  none: 0,
  default: '0 1px 1px 0 rgba(0, 0, 0, 0.1)',
  heavy: '0 2px 2px 0 rgba(0, 0, 0, 0.1)',
  focus: `0 0 2px 2px ${colors.greys.dusty};`,
};

const space = range(0, 129, 4).map(i => rem(`${i}px`));

const textStyles = {
  caps: {
    textTransform: 'uppercase',
    fontWeight: fontWeights.bold,
  },

  h1: {
    fontSize: fontSizes['3xl'],
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.normal,
  },
  h2: {
    fontSize: fontSizes['2xl'],
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.normal,
  },
  h3: {
    fontSize: fontSizes.xl,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h4: {
    fontSize: fontSizes.lg,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h5: {
    fontSize: fontSizes.base,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h6: {
    fontSize: fontSizes.sm,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  label: {
    fontSize: fontSizes.sm,
    lineHeight: lineHeights.loose,
    fontWeight: fontWeights.bold,
    color: colors.greys.charcoal,
  },
  text: {
    fontSize: fontSizes.base,
    lineHeight: lineHeights.loose,
    fontWeight: fontWeights.normal,
  },
};

const transitions = {
  default: '200ms ease-in',
};

export default {
  alertStyles,
  borders,
  breakpoints,
  buttons,
  colors,
  fontFamily,
  fontSizes,
  fontWeights,
  icons,
  letterSpacings,
  lineHeights,
  maxWidths,
  mediaQueries,
  opacity,
  radii,
  shadows,
  space,
  textStyles,
  transitions,
};
