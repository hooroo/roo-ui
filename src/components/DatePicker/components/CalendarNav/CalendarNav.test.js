import React from 'react';
import theme from 'theme';
import { mountWithTheme } from 'testUtils';

import CalendarNav from './';

describe('<CalendarNav />', () => {
  let wrapper;

  const props = {
    prevProps: { onClick: jest.fn() },
    nextProps: { onClick: jest.fn() },
  };

  beforeEach(() => {
    wrapper = mountWithTheme(<CalendarNav {...props} />, theme);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('passes props.prevProps to first <Button />', () => {
    expect(
      wrapper
        .find('Button')
        .first()
        .props(),
    ).toEqual(expect.objectContaining(props.prevProps));
  });

  it('passes props.nextProps to second <Button />', () => {
    expect(
      wrapper
        .find('Button')
        .last()
        .props(),
    ).toEqual(expect.objectContaining(props.nextProps));
  });
});
