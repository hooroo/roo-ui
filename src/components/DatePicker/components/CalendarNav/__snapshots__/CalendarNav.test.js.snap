// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarNav /> renders correctly 1`] = `
.emotion-22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  position: absolute;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0;
}

.emotion-9 {
  border: none;
  width: auto;
  overflow: visible;
  font: inherit;
  line-height: normal;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  margin: 0;
  padding: 0;
  color: inherit;
  background-color: transparent;
}

.emotion-9:disabled {
  cursor: not-allowed;
}

.emotion-9:hover:disabled {
  color: inherit;
  background-color: transparent;
}

.emotion-9:hover,
.emotion-9:focus {
  outline: none;
}

.emotion-9:disabled {
  cursor: not-allowed;
  box-shadow: none;
}

.emotion-3 {
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  vertical-align: middle;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

<CalendarNav
  nextProps={
    Object {
      "onClick": [MockFunction],
    }
  }
  prevProps={
    Object {
      "onClick": [MockFunction],
    }
  }
>
  <Wrapper>
    <div
      className="emotion-22 emotion-23"
    >
      <Button
        as="button"
        bg="transparent"
        color="inherit"
        m={0}
        onClick={[MockFunction]}
        p={0}
        type="button"
      >
        <button
          className="emotion-9 emotion-10"
          color="inherit"
          onClick={[MockFunction]}
          type="button"
        >
          <Icon
            name="chevronLeft"
            size={24}
            title={null}
          >
            <WithTheme(Component)
              className="emotion-3 emotion-0"
              name="chevronLeft"
              size={24}
              title={null}
            >
              <Component
                className="emotion-3 emotion-0"
                name="chevronLeft"
                size={24}
                title={null}
              >
                <StyledSvg
                  className="emotion-3 emotion-0"
                  fill="currentcolor"
                  height={24}
                  title="chevronLeft"
                  viewBox="0 0 24 24"
                  width={24}
                >
                  <svg
                    className="emotion-0 emotion-1 emotion-2"
                    fill="currentcolor"
                    height={24}
                    title="chevronLeft"
                    viewBox="0 0 24 24"
                    width={24}
                  >
                    <path
                      d={null}
                    />
                  </svg>
                </StyledSvg>
              </Component>
            </WithTheme(Component)>
          </Icon>
        </button>
      </Button>
      <Button
        as="button"
        bg="transparent"
        color="inherit"
        m={0}
        onClick={[MockFunction]}
        p={0}
        type="button"
      >
        <button
          className="emotion-9 emotion-10"
          color="inherit"
          onClick={[MockFunction]}
          type="button"
        >
          <Icon
            name="chevronRight"
            size={24}
            title={null}
          >
            <WithTheme(Component)
              className="emotion-3 emotion-0"
              name="chevronRight"
              size={24}
              title={null}
            >
              <Component
                className="emotion-3 emotion-0"
                name="chevronRight"
                size={24}
                title={null}
              >
                <StyledSvg
                  className="emotion-3 emotion-0"
                  fill="currentcolor"
                  height={24}
                  title="chevronRight"
                  viewBox="0 0 24 24"
                  width={24}
                >
                  <svg
                    className="emotion-0 emotion-1 emotion-2"
                    fill="currentcolor"
                    height={24}
                    title="chevronRight"
                    viewBox="0 0 24 24"
                    width={24}
                  >
                    <path
                      d={null}
                    />
                  </svg>
                </StyledSvg>
              </Component>
            </WithTheme(Component)>
          </Icon>
        </button>
      </Button>
    </div>
  </Wrapper>
</CalendarNav>
`;
